semgrep-lint crashes if /tmp doesn't exist

**Describe the bug**

I think this line is where we expect `/tmp` to exist https://github.com/returntocorp/semgrep/commit/bc39236b35c99eea436228189eff5f4097266abd#diff-2c8973c1a63d95e440b7c99e07c47b7dR189 and semgrep-lint crashes if it doesn't.

**To Reproduce**

1. Make sure there's no `/tmp`
2. Run `sgrep` tool via bento

**Expected behavior**

semgrep should use `/var/tmp` in docker contexts where that path is used instead of `/tmp`

**Screenshots**

```
[INFO] 5460.999965667725 root:docker - Running sgrep using installed binary
[INFO] 5701.299667358398 root:docker - sgrep: Returned code 2 with stdout[:4000]:
stderr[:4000]
Fatal error: exception (Failure
  "sgrep_lint: wrong format. could not parse the pattern $X == $X (exn = (Sys_error \"/tmp/tmp-431-508804.py: No such file or directory\"))")
non-zero return code while invoking sgrep with:
	sgrep -lang python -rules_file /var/tmp/tmp15_abs8h /src/bento.py
Command '['sgrep', '-lang', 'python', '-rules_file', '/var/tmp/tmp15_abs8h', '/src/bento.py']' returned non-zero exit status 2.
An error occurred while invoking the sgrep engine; please help us fix this by filing an an issue at https://sgrep.dev
[ERROR] 5863.148212432861 root:tool_runner - Traceback (most recent call last):
  File "/app/bento/tool_runner.py", line 257, in _setup_and_run_single_tool
    results = self._run_single_tool(bar, ix, tool, baseline)
  File "/app/bento/tool_runner.py", line 226, in _run_single_tool
    tool.tool_id(), tool.results(self.paths, self.use_cache), baseline
  File "/app/bento/tool/tool.py", line 269, in results
    violations = self._get_findings_from_run(paths)
  File "/app/bento/tool/tool.py", line 234, in _get_findings_from_run
    raw = self.run(path_list)
  File "/app/bento/extra/base_sgrep.py", line 122, in run
    output_str = self.run_command(files).stdout.split("\n")
  File "/app/bento/tool/runner/docker.py", line 247, in run_command
    raise subprocess.CalledProcessError(
subprocess.CalledProcessError: Command '['--config', '.bento/sgrep.yml', '--json', '--skip-pattern-validation']' returned non-zero exit status 2.
```


**Environment**

returntocorp/bento docker image

--- a/semgrep/semgrep/config_resolver.py
+++ b/semgrep/semgrep/config_resolver.py
@@ -3,6 +3,7 @@ import os
 import shutil
 import sys
 import tarfile
+import tempfile
 import time
 from pathlib import Path
 from typing import Any
@@ -182,14 +183,14 @@ def download_config(config_url: str) -> Dict[str, Optional[Dict[str, Any]]]:
             if content_type and "text/plain" in content_type:
                 return parse_config_string("remote-url", r.content.decode("utf-8"))
             elif content_type and content_type == "application/x-gzip":
-                fname = f"/tmp/{base64.b64encode(config_url.encode()).decode()}"
-                shutil.rmtree(fname, ignore_errors=True)
-                with tarfile.open(fileobj=r.raw, mode="r:gz") as tar:
-                    tar.extractall(fname)
-                extracted = Path(fname)
-                for path in extracted.iterdir():
-                    # get first folder in extracted folder (this is how GH does it)
-                    return parse_config_folder(path, relative=True)
+                directory_name = base64.b64encode(config_url.encode()).decode()
+                with tempfile.TemporaryDirectory(suffix=directory_name) as fname:
+                    with tarfile.open(fileobj=r.raw, mode="r:gz") as tar:
+                        tar.extractall(fname)
+                    extracted = Path(fname)
+                    for path in extracted.iterdir():
+                        # get first folder in extracted folder (this is how GH does it)
+                        return parse_config_folder(path, relative=True)
             else:
                 print_error_exit(
                     f"unknown content-type: {content_type} returned by config url: {config_url}. Can not parse"

--- a/semgrep/semgrep/config_resolver.py
+++ b/semgrep/semgrep/config_resolver.py
@@ -183,8 +183,7 @@ def download_config(config_url: str) -> Dict[str, Optional[Dict[str, Any]]]:
             if content_type and "text/plain" in content_type:
                 return parse_config_string("remote-url", r.content.decode("utf-8"))
             elif content_type and content_type == "application/x-gzip":
-                directory_name = base64.b64encode(config_url.encode()).decode()
-                with tempfile.TemporaryDirectory(suffix=directory_name) as fname:
+                with tempfile.TemporaryDirectory() as fname:
                     with tarfile.open(fileobj=r.raw, mode="r:gz") as tar:
                         tar.extractall(fname)
                     extracted = Path(fname)

